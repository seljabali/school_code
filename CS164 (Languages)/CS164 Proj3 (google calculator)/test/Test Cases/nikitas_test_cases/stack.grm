//a stack language
%%
ARPN -> V ARPN | UNIT ARPN | _ ;
V -> V V OP | ID | NUM;
UNIT -> ASGN | RM;
ASGN -> V ID ':=';
RM -> ID 'x=';
OP -> '+' | 'x' | '-';
ID -> /[a-zA-Z][a-zA-Z0-9]*/;
NUM -> /[1-9][0-9]*/;