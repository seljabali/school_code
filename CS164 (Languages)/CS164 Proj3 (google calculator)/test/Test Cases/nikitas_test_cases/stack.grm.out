(Grammar
    (Declarations
    )
    (Rules
        (Rule ARPN
            (production [ V, ARPN, ]
            )
            (production [ UNIT, ARPN, ]
            )
            (production [ (epsilon), ]
            )
        )
        (Rule V
            (production [ V, V, OP, ]
            )
            (production [ ID, ]
            )
            (production [ NUM, ]
            )
        )
        (Rule UNIT
            (production [ ASGN, ]
            )
            (production [ RM, ]
            )
        )
        (Rule ASGN
            (production [ V, ID, \:\=, ]
            )
        )
        (Rule RM
            (production [ ID, x\=, ]
            )
        )
        (Rule OP
            (production [ \+, ]
            )
            (production [ x, ]
            )
            (production [ \-, ]
            )
        )
        (Rule ID
            (production [ [a-zA-Z][a-zA-Z0-9]*, ]
            )
        )
        (Rule NUM
            (production [ [1-9][0-9]*, ]
            )
        )
    )
    (StartSymbol ARPN)
)
